src/Navigation/Dropdown/Dropdown.test.tsx

```diff
@@ -346,3 +423,86 @@ describe('<Dropdown/> logic', () => {
     expect(onChange.mock.calls[0][0]).toEqual('pm');
   });
 });
+
+describe('<DropdownItem/> onClick', () => {
+  test('onClick callback should be called when it is a function', async () => {
+    const onClick = jest.fn();
+    const menuText = 'Product Manager';
+    render(
+      <Dropdown label="Career">
+        <DropdownItem value="pm" onClick={onClick}>
+          {menuText}
+        </DropdownItem>
+      </Dropdown>
+    );
+
+    const dropdown = queryByRole(document.body, 'menuitem');
+    fireEvent.click(dropdown);
+    const option = queryByText(document.body, menuText);
+    fireEvent.mouseDown(option);
+
+    expect(onClick).toHaveBeenCalledTimes(1);
+  });
+
+  test('invalid onClick value should not break anything', async () => {
+    const onClick = 'invalid-on-click';
+    const menuText = 'Product Manager';
+    render(
+      <Dropdown label="Career">
+        <DropdownItem
+          value="pm"
+          // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+          // @ts-ignore
+          onClick={onClick}
```

Comment:
What was the problem here?

Reply:
This test case is to test when the onClick is not a function type, it won't break the website.

So I made the `onClick` be a string type which gets TS to complain about it.

Reply:
Oooh lol. Okay makes sense. Though I kinda disagree with it not breaking when onClick is not a string. Well, out of scope I guess.